### PR TITLE
feat : added playlist

### PR DESCRIPTION
**Pull request by Sharmaryan**
- user can create playlists
- user can remove videos from a playlist

### CODE REVIEW

**Review by shashikantmaurya0007 — APPROVED** [review]
good folder structure
great how you created sidebar outside and rendered childeren as outlet
just work on uI rest is good

**Review by sonishreyas — APPROVED** [review]
Code looks good. Easy to understand

File: src/backend/controllers/PlaylistController.js [hunk 1]
```diff
@@ -89,7 +89,9 @@ export const getVideosFromPlaylistHandler = function (schema, request) {
   const user = requiresAuth.call(this, request);
   if (user) {
     const playlistId = request.params.playlistId;
-    const playlist = user.playlists.find((item) => item._id !== playlistId);
+    const playlist = user.playlists.find((item) => item._id === playlistId);
+    console.log('playlist coming from backend',playlist)
```
Comment (TejasShekar) [inline src/backend/controllers/PlaylistController.js]: Remove if not needed

File: src/components/VideoSinglePlaylist/VideoSinglePlaylist.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  const { singlePlayList, setSinglePlayList } = useSinglePlayList();
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = await axios.get(`/api/user/playlists/${playlistId}`, {
+          headers: { authorization: auth.token },
+        });
+        setSinglePlayList(response.data.playlist.videos);
+        console.log(response);
```
Comment (TejasShekar) [inline src/components/VideoSinglePlaylist/VideoSinglePlaylist.js]: remove console.log

File: src/components/VideosPlaylists/VideosPlaylists.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  }, [auth.token, setPlayLists]);
+
+  return (
+    <div className="videos-playlists">
+      { playLists?.map((playlist) => {
+            const {title, _id, videos} = playlist;
+          return (
+            <div className="playlist-card" key={_id}>
+              <Link to={`/playlists/${_id}`}>
+                {console.log(playlist.videos[0]?.thumbnail)}
```
Comment (TejasShekar) [inline src/components/VideosPlaylists/VideosPlaylists.js]: remove console.log

File: src/services/playlist.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        console.log(err);
+      }
+    })();
+  } else {
+    (async () => {
+      try {
+        const response = await axios.delete(`/api/user/playlists/${_id}`, {
+          headers: { authorization: auth.token },
+        });
+        console.log(response);
```
Comment (TejasShekar) [inline src/services/playlist.js]: remove if not needed.

**Review by TejasShekar — APPROVED** [review]
UI :
- Homepage banner - use object-fit on the image and position it.
- The selected video page will off-center to the right of the screen.
- Make categories clickable.
- Add cursor pointer to buttons.
- Take care of creating a playlist with no name provided.
- Add a redirect file to fix netlify broken link issue

**Pull request merged**